web/modules/custom/bnf/bnf_client/src/Form/BnfSyncForm.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Drupal\bnf_client\Form;
+
+use Drupal\bnf_client\BnfScheduler;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\DependencyInjection\DependencySerializationTrait;
+use Drupal\Core\Form\FormInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\Core\StringTranslation\TranslationInterface;
+use Drupal\queue_ui\QueueUIBatchInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Full sync form.
+ */
+class BnfSyncForm implements FormInterface, ContainerInjectionInterface {
+
+  use DependencySerializationTrait;
+  use StringTranslationTrait;
+
+  /**
+   * Constructor.
+   */
+  public function __construct(
+    protected BnfScheduler $scheduler,
+    protected QueueUIBatchInterface $queueBatch,
+    TranslationInterface $stringTranslation,
+  ) {
+    $this->setStringTranslation($stringTranslation);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function create(ContainerInterface $container): self {
+    return new static(
+      $container->get(BnfScheduler::class),
+      $container->get('queue_ui.batch'),
+      $container->get('string_translation'),
+    );
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function getFormId(): string {
+    return 'bnf_import_sync_form';
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+    $form['#title'] = $this->t('Refresh all subscriptions?', [], ['context' => 'BNF']);
```

Comment:
Please add a `#description` here also, so the user of the form knows what refresh actually means

Without having the background knowledge, It could also mean that all the existing content gets deleted :)